src/ResultWriter/EnergyOutput.cpp

```diff
@@ -613,35 +616,45 @@ void EnergyOutput::printEnergies() {
     const auto totalMomentumY = energiesStorage.totalMomentumY();
     const auto totalMomentumZ = energiesStorage.totalMomentumZ();
 
+    const auto outputPrecision =
+        seissolInstance.getSeisSolParameters().output.energyParameters.terminalPrecision;
+
     if (shouldComputeVolumeEnergies()) {
       if (totalElasticEnergy) {
-        logInfo(rank) << "Elastic energy (total, % kinematic, % potential): " << totalElasticEnergy
+        logInfo(rank) << std::setprecision(outputPrecision)
+                      << "Elastic energy (total, % kinematic, % potential): " << totalElasticEnergy
                       << " ," << ratioElasticKinematic << " ," << ratioElasticPotential;
       }
       if (totalAcousticEnergy) {
-        logInfo(rank) << "Acoustic energy (total, % kinematic, % potential): "
+        logInfo(rank) << std::setprecision(outputPrecision)
+                      << "Acoustic energy (total, % kinematic, % potential): "
                       << totalAcousticEnergy << " ," << ratioAcousticKinematic << " ,"
                       << ratioAcousticPotential;
       }
       if (energiesStorage.gravitationalEnergy()) {
-        logInfo(rank) << "Gravitational energy:" << energiesStorage.gravitationalEnergy();
+        logInfo(rank) << std::setprecision(outputPrecision)
+                      << "Gravitational energy:" << energiesStorage.gravitationalEnergy();
       }
       if (energiesStorage.plasticMoment()) {
-        logInfo(rank) << "Plastic moment (value, equivalent Mw, % total moment):"
+        logInfo(rank) << std::setprecision(outputPrecision)
+                      << "Plastic moment (value, equivalent Mw, % total moment):"
                       << energiesStorage.plasticMoment() << " ,"
                       << 2.0 / 3.0 * std::log10(energiesStorage.plasticMoment()) - 6.07 << " ,"
                       << ratioPlasticMoment;
       }
-      logInfo(rank) << "Total momentum (X, Y, Z):" << totalMomentumX << " ," << totalMomentumY
+      logInfo(rank) << std::setprecision(outputPrecision)
+                    << "Total momentum (X, Y, Z):" << totalMomentumX << " ," << totalMomentumY
                     << " ," << totalMomentumZ;
     } else {
-      logInfo(rank) << "Volume energies skipped at this step";
+      logInfo(rank) << std::setprecision(outputPrecision) << "Volume energies skipped at this step";
```

Comment:
Why do we have a precision for just the text output?

Reply:
For no reason. :) It's removed.